ledger refactoring: make consensus upgrade

tsachiherman opened the pull request:
## Summary

Create consensus upgrade to V32

## Test Plan

Use existing unit tests.

algorandskiy commented on ledger/applications_test.go:
```diff
@@ -134,18 +134,34 @@ return`
-	expectedUserLocalResource, err := hex.DecodeString("82a16f01a17081a26c6b82a27462a56c6f63616ca2747401")
-	a.NoError(err)
+	var expectedCreatorBase, expectedCreatorResource, expectedUserOptInBase, expectedUserOptInResource, expectedUserLocalBase, expectedUserLocalResource []byte
+	if proto.EnableAccountDataResourceSeparation {
```
please update the test comment explaining what changed this time in these hardcoded values

tsachiherman replied:
added

algorandskiy reviewed (changes requested):
The consensus upgrade looks good, the test needs a comment

cce commented on config/consensus.go:
```diff
@@ -1083,38 +1083,46 @@ func initConsensusProtocols() {
+	vFuture.ApprovedUpgrades = map[protocol.ConsensusVersion]uint64{}
+
+	// FilterTimeout for period 0 should take a new optimized, configured value, need to revisit this later
+	vFuture.AgreementFilterTimeoutPeriod0 = 4 * time.Second
```
we have always just pushed this forward, and I guess this is like a TODO, but this line doesn't do anything since the base version is also 4 seconds.

algorandskiy reviewed (approved):
Why did the test change go?

tsachiherman commented:
> Why did the test change go?

I extracted the test change from this PR, and moved it to the feature branch. That way, this PR is just the consensus changes.
